### PR TITLE
re-add custom core-audio truncation after syncing fork

### PR DESCRIPTION
**Pull request by lkotsonis**

### CODE REVIEW

**Review by dgrahelj — APPROVED** [review]
I didn't know this was still used. Does any part of our stack still require bitperfect float<->int conversions?

**Comment by lkotsonis** [comment]
> I didn't know this was still used. Does any part of our stack still require bitperfect float<->int conversions?

I guess the question is if any part uses CoreAudio conversions... @wiccy46 are we doing any audio tests using CoreAudio/mac devices?

File: RtAudio.cpp [hunk 1]
```diff
@@ -11210,8 +11210,8 @@ void RtApi :: convertBuffer( char *outBuffer, char *inBuffer, ConvertInfo &info
       Float32 *in = (Float32 *)inBuffer;
       for (unsigned int i=0; i<stream_.bufferSize; i++) {
         for (j=0; j<info.channels; j++) {
-          // Use llround() which returns `long long` which is guaranteed to be at least 64 bits.
-          out[info.outOffset[j]] = (Int32) std::max(std::min(std::llround(in[info.inOffset[j]] * 2147483648.f), 2147483647LL), -2147483648LL);
+          long long value = in[info.inOffset[j]] * 2147483648LL;
+          out[info.outOffset[j]] = std::max(-2147483648LL, std::min(value, 2147483647LL));
```
Comment (actondev) [inline RtAudio.cpp:11214]: nit: std::clamp

**Review by actondev — COMMENTED** [review]
nit: std::clamp to all max(.., min(..))

**Pull request merged**